### PR TITLE
Alt fix pdf delivery

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality or fixes a bug.

I mostly wanted to make sure I understood the issue. Your solution is easier, so we can go with it as well! It just feels a little annoying to have hard coded file ending usage, when we anyway have a format tag in the describing dict. Making a reverse lookup dict for this key is a little annoying but cheap. 

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

**Comment by dnil** [comment]
I feel the risk of the ego arguing here, but I'd say the test fix at least should be added: it makes sense to use the test with the parameter it would be passed when actually using the endpoint. Ego aside, it kind of still works out either way, as both test versions actually produce a delivery report pdf.

File: scout/constants/case_tags.py [hunk 1]
```diff
@@ -33,6 +33,13 @@
     },
 }
 
+CUSTOM_CASE_REPORT_KEY_NAMES = dict(
```
Comment (northwestwitch) [inline scout/constants/case_tags.py:36]: For me it is a bit hard to understand and follow what the field of this variable are. What is it, it's very nested in the end

File: scout/server/blueprints/cases/views.py [hunk 2]
```diff
@@ -725,7 +724,11 @@ def custom_report(institute_id, case_name, report_type):
         try:
             with open(os.path.abspath(report_path), "r") as html_file:
                 source_code = html_file.read()
-                if CUSTOM_CASE_REPORTS[report_type]["format"] == "YAML":
+
+                if (
+                    CUSTOM_CASE_REPORTS[CUSTOM_CASE_REPORT_KEY_NAMES[report_type]]["format"]
```
Comment (northwestwitch) [inline scout/server/blueprints/cases/views.py:729]: This works, but the syntax is quite complex with a very nested dictionary. I don't like much my solution but I find it simpler to follow and to maintain?

**Review by northwestwitch — COMMENTED** [review]
Check my comments!

File: scout/server/blueprints/cases/views.py (hunk 2 above)
Comment (dnil) [reply scout/server/blueprints/cases/views.py:729]: I prefer this one, but I'm ok with your solution just checking file ending. But I would prefer it if we at least also fix the endpoint test then to use the keys the endpoint actually receives...

Comment (northwestwitch) [reply scout/server/blueprints/cases/views.py:729]: I can add some parametrized tests tomorrow!

Comment (dnil) [reply scout/server/blueprints/cases/views.py:729]: It has a parametrised test already; just needs the proper key - but you can e.g. reuse the reverse dict from here, or make a list directly from the original dict member dicts key_name members!

Comment (northwestwitch) [reply scout/server/blueprints/cases/views.py:729]: But Ok, I'll check the code tomorrow!

**Comment by dnil** [comment]
The test is now fixed in the parent PR as well, so closing this. While I still think a reverse lookup was not a bad two-with-one-stone solution, checking filename endings explicitly and skipping the test parametrisation for an explicit loop also works.

**Comment by northwestwitch** [comment]
> The test is now fixed in the parent PR as well, so closing this. While I still think a reverse lookup was not a bad two-with-one-stone solution, checking filename endings explicitly and skipping the test parametrisation for an explicit loop also works

After the third level of dictionary nesting it should be considered machine code 😄

**Pull request closed without merging**